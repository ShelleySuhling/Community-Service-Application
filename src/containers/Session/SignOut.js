import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as sessionActions from '../../actions/sessionActions';
import PropTypes from 'prop-types';

class SignOut extends Component {

    constructor(props) {
        super(props)
        this.state = {
            loading: this.props.session.pending_signout
        }
    }

    componentWillMount() {
        this.props.sessionActions.requestSignOut(this.state)
    }

    render() {
        return (
            < div className="content-container" >
                {this.state.loading ? "Signing you out" : "You've been successfully signed out"}
            </div >
        )
    }
}

SignOut.propTypes = {
    sessionActions: PropTypes.object,
    session: PropTypes.object
};
function mapStateToProps(state) {
    return {
        session: state.session
    };
}
function mapDispatchToProps(dispatch) {
    return {
        sessionActions: bindActionCreators(sessionActions, dispatch)
    };
}
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(SignOut);