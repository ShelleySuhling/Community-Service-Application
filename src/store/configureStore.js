import { applyMiddleware, createStore } from 'redux'
import { persistStore, persistCombineReducers } from 'redux-persist';
import thunkMiddleware from 'redux-thunk'
import storage from 'redux-persist/lib/storage';

import rootReducer from '../reducers/rootReducer'

export default function configureStore(preloadedState) {

    const persistConfig = {
        key: 'root',
        storage: storage,
    };
    const pReducer = persistCombineReducers(persistConfig, rootReducer);

    const store = createStore(pReducer, applyMiddleware(thunkMiddleware));
    const persistor = persistStore(store);
    window.store = store;

    return { persistor, store }
}