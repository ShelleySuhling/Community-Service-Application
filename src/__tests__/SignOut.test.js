import * as React from 'react';
import { shallow } from 'enzyme';

import SignOut from '../containers/Session/SignOut';

import configureStore from '../store/configureStore';
const { store } = configureStore()

describe('SignOut', () => {
    const signOutWrapper = shallow(<SignOut store={store} />).dive()

    it('matches snapshot', () => {
        expect(signOutWrapper).toMatchSnapshot();
    })
    it('mounts and requests sign out', () => {
        const request_sign_out_mock = jest.fn()
        signOutWrapper.instance().props.sessionActions.requestSignOut = request_sign_out_mock

        signOutWrapper.instance().componentWillMount()
        expect(request_sign_out_mock).toHaveBeenCalled()
    })
    it('correctly displays loading screen', () => {
        signOutWrapper.instance().state.loading = true
        signOutWrapper.instance().forceUpdate()
        expect(signOutWrapper.text()).toEqual('Signing you out')
    })
    it('correctly displays successful sign out', () => {
        signOutWrapper.instance().state.loading = false
        signOutWrapper.instance().forceUpdate()
        expect(signOutWrapper.text()).toEqual('You\'ve been successfully signed out')
    })
})