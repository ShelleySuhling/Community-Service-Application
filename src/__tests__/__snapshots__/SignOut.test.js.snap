// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOut matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignOut
    session={Object {}}
    sessionActions={
      Object {
        "clearAuthErrors": [Function],
        "requestSignIn": [Function],
        "requestSignOut": [Function],
        "requestSignUp": [Function],
        "updateSessionUser": [Function],
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "You've been successfully signed out",
      "className": "content-container",
    },
    "ref": null,
    "rendered": "You've been successfully signed out",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "You've been successfully signed out",
        "className": "content-container",
      },
      "ref": null,
      "rendered": "You've been successfully signed out",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
